FabLabKasse/faucardPayment/generateLog.py

```diff
@@ -126,7 +144,7 @@ def verify_sum(curKb, start, end, magposSum):
                 lastdate = timestamp
 
             # Determine corresponding rechnung
-            curKb.execute("SELECT rechnung FROM buchung WHERE konto = 'FAUKarte' AND (betrag = ? OR betrag = ?) AND datum BETWEEN ? AND ? ",(unicode(row[3]), "{:.3f}".format(row[3]), timestamp, timestamp + timedelta(seconds=20)))
+            curKb.execute("SELECT rechnung, datum FROM buchung WHERE konto = 'FAUKarte' AND (betrag = ? OR betrag = ? OR betrag = ? OR betrag = ? OR betrag = ? OR betrag = ?) AND datum BETWEEN ? AND ? ",(unicode(amount),"{:.0f}".format(amount),"{:.1f}".format(amount), "{:.2f}".format(amount), "{:.3f}".format(amount), "{:.4f}".format(amount), timestamp, timestamp + timedelta(seconds=20)))
```

Comment:
Do we really need the OR query? For float, `abs(betrag - ?) < 1e-4` should work as well, and for text, we should normalize when saving.